### DISCUSSION TITLE
Cannot persist a StringProperty

### DISCUSSION THREAD

**Discussion by SR-Lut3t1um** [microstream-one/microstream#243] [Q&A]
Hello,
I'm running into an issue when using JavaFX, where I'm trying to persist a StringProperty that has the default listeners added via adding them to a table. However since the invalidation listener is a weak reference, which cannot be persisted I'm running into a exception, that makes persisting it impossible. I tried to create a custom BinaryTypeHandler however it does not have a Binary_String method. Am I missing something here or would the List in wich JavaFX stores the handler need to be transient. 
I've also made a simple [recreater](https://github.com/SR-Lut3t1um/recreaterMicrostreamJavaFX).
Any help would be highly appreciated! Kindest regards.

**Comment by hg-ms** [comment]
Hello,
As the JavaFX Properties are quite complex classes I’d rather not try to persist them directly. Maybe you can create a data-layer that is independent of the JavaFX classes and persist that.

Nevertheless here are some hints:
Storing Strings:
Strings are stored as character arrays. (see one.microstream.persistence.binary.java.lang.BinaryHandlerString)
the method you are looking for may be one.microstream.persistence.binary.types.Binary.storeStringSingleValue(long, long, char[], int, int)

Ignoring Fields:
If it is about ignoring a non-transient field, you may use custom PersistenceFieldEvaluator: 

```
public class CustomPersistenceFieldEvaluator implements PersistenceFieldEvaluator {

	@Override
	public boolean applies(final Class<?> entityType, final Field field) {

		//return true if field should be persisted, false if not
		…

		//default: return false if field has the transient modifier
		return !XReflect.isTransient(field);
	}

}
```
see https://docs.microstream.one/manual/storage/storing-data/transient-fields.html
